Issue/22

coldrice99 opened the pull request:
게시물 작성 기능 구현

coldrice99 committed:
feat: 게시물 작성 API 구현 및 관련 DTO 추가

- 게시물 생성 기능 추가 (PostService 및 PostController 작성)
- 요청 데이터 유효성 검사 및 예외 처리 포함
- PostRequestDto, PostResponseDto 추가
- CommentRequestDto, CommentResponseDto 추가
- 댓글 관련 DTO 추가로 인해 다른 팀원과의 충돌 방지 목적

coldrice99 committed:
feat: 게시물 작성 API 구현 및 관련 DTO 추가

- 게시물 생성 기능 추가 (PostService 및 PostController 작성)
- 요청 데이터 유효성 검사 및 예외 처리 포함
- PostRequestDto, PostResponseDto 추가
- CommentRequestDto, CommentResponseDto 추가
- 댓글 관련 DTO 추가로 인해 다른 팀원과의 충돌 방지 목적

coldrice99 committed:
feat: 게시물 작성 API 구현 및 관련 DTO 추가

- 게시물 생성 기능 추가 (PostService 및 PostController 작성)
- 요청 데이터 유효성 검사 및 예외 처리 포함
- PostRequestDto, PostResponseDto 추가
- CommentRequestDto, CommentResponseDto 추가
- 댓글 관련 DTO 추가로 인해 다른 팀원과의 충돌 방지 목적

coldrice99 committed:
feat: 게시물 작성 API 구현 및 관련 DTO 추가

- 게시물 생성 기능 추가 (PostService 및 PostController 작성)
- 요청 데이터 유효성 검사 및 예외 처리 포함
- PostRequestDto, PostResponseDto 추가
- CommentRequestDto, CommentResponseDto 추가
- 댓글 관련 DTO 추가로 인해 다른 팀원과의 충돌 방지 목적

coldrice99 committed:
feat: 게시물 수정 기능 구현

- 게시물 수정 API 추가 (PostService 및 PostController 작성)
- 작성자 검증 로직 추가로 권한 없는 사용자의 수정 방지
- PostRequestDto, PostResponseDto를 사용하여 요청 데이터와 응답 데이터 처리
- 서비스 계층에서 게시물 수정 후 업데이트된 데이터 반환

coldrice99 committed:
feat: 게시물 수정 기능 구현

- 게시물 수정 API 추가 (PostService 및 PostController 작성)
- 작성자 검증 로직 추가로 권한 없는 사용자의 수정 방지
- PostRequestDto, PostResponseDto를 사용하여 요청 데이터와 응답 데이터 처리
- 서비스 계층에서 게시물 수정 후 업데이트된 데이터 반환

coldrice99 committed:
feat: 게시물 작성 기능 구현

- 게시물 작성 API 추가 (PostService 및 PostController 작성)
- JWT 인증을 통해 작성자 정보 확인 및 게시물 작성 가능
- Member 엔티티의 lazy-loading 문제를 해결하기 위해 MemberResponseDto를 사용하여 불필요한 데이터 로딩과 직렬화 문제 방지
- PostRequestDto, PostResponseDto를 사용하여 요청 데이터와 응답 데이터 처리

issue #22

coldrice99 committed:
feat: 게시물 작성 기능 구현

- 게시물 작성 API 추가 (PostService 및 PostController 작성)
- JWT 인증을 통해 작성자 정보 확인 및 게시물 작성 가능
- Member 엔티티의 lazy-loading 문제를 해결하기 위해 MemberResponseDto를 사용하여 불필요한 데이터 로딩과 직렬화 문제 방지
- PostRequestDto, PostResponseDto를 사용하여 요청 데이터와 응답 데이터 처리

issue #22

coldrice99 committed:
Merge remote-tracking branch 'origin/issue/22' into issue/22

# Conflicts:
#	src/main/java/com/sparta/newsfeedproject/domain/comment/dto/CommentRequestDto.java
#	src/main/java/com/sparta/newsfeedproject/domain/comment/dto/CommentResponseDto.java
#	src/main/java/com/sparta/newsfeedproject/domain/comment/entity/Comment.java
#	src/main/java/com/sparta/newsfeedproject/domain/member/repository/MemberRepository.java
#	src/main/java/com/sparta/newsfeedproject/domain/post/controller/PostController.java
#	src/main/java/com/sparta/newsfeedproject/domain/post/dto/PostRequestDto.java
#	src/main/java/com/sparta/newsfeedproject/domain/post/dto/PostResponseDto.java
#	src/main/java/com/sparta/newsfeedproject/domain/post/entity/Post.java
#	src/main/java/com/sparta/newsfeedproject/domain/post/repository/PostRepository.java
#	src/main/java/com/sparta/newsfeedproject/domain/post/service/PostService.java
#	src/main/resources/application.properties

coldrice99 committed:
feat: 게시물 작성 기능 구현

- 게시물 작성 API 추가 (PostService 및 PostController 작성)
- JWT 인증을 통해 작성자 정보 확인 및 게시물 작성 가능
- Member 엔티티의 lazy-loading 문제를 해결하기 위해 MemberResponseDto를 사용하여 불필요한 데이터 로딩과 직렬화 문제 방지
- PostRequestDto, PostResponseDto를 사용하여 요청 데이터와 응답 데이터 처리

issue #22

coldrice99 committed:
feat: 게시물 작성 기능 구현

- 게시물 작성 API 추가 (PostService 및 PostController 작성)
- JWT 인증을 통해 작성자 정보 확인 및 게시물 작성 가능
- Member 엔티티의 lazy-loading 문제를 해결하기 위해 MemberResponseDto를 사용하여 불필요한 데이터 로딩과 직렬화 문제 방지
- PostRequestDto, PostResponseDto를 사용하여 요청 데이터와 응답 데이터 처리

issue #22

coldrice99 committed:
refactor: MemberController 삭제

- Post 폴더에 테스트용으로 만들었던 MemberController 클래스를 삭제하여 클래스 중복 문제 개선

Refs: #25

coldrice99 committed:
refactor: MemberController 삭제

- Post 폴더에 테스트용으로 만들었던 MemberController 클래스를 삭제하여 클래스 중복 문제 개선

Refs: #25

coldrice99 committed:
feat: 불필요한 코드 제거

- 중복되거나 사용되지 않는 코드 세그먼트를 정리하여 코드베이스를 정돈했습니다.
- 프로젝트의 가독성과 유지보수성을 향상시켰습니다.

Refs: #22

coldrice99 committed:
feat: 불필요한 코드 제거

- 중복되거나 사용되지 않는 코드 세그먼트를 정리하여 코드베이스를 정돈했습니다.
- 프로젝트의 가독성과 유지보수성을 향상시켰습니다.

Refs: #22

coldrice99 committed:
Merge remote-tracking branch 'origin/issue/22' into issue/22

inzulmi132 commented on src/main/java/com/sparta/newsfeedproject/domain/comment/entity/Comment.java:
```diff
@@ -23,12 +25,12 @@ public class Comment extends TimeStamped {
     // 게시글과 댓글 = 다 대 일
     @ManyToOne(fetch = FetchType.LAZY)
-    @JoinColumn(name = "post_id", nullable = false)
+    @JoinColumn(name = "post_id",nullable = false)
```
쉼표 다음에는 띄어쓰기가 들어가야 합니다.

inzulmi132 commented on src/main/java/com/sparta/newsfeedproject/domain/comment/entity/Comment.java:
```diff
@@ -6,9 +6,11 @@
 
 @Entity
 @Getter
+@Setter
```
사용하지 않는 어노테이션이면 빼는 게 좋을 거 같습니다.

inzulmi132 commented on src/main/java/com/sparta/newsfeedproject/domain/post/dto/PostResponseDto.java:
```diff
@@ -0,0 +1,32 @@
+    private Long id;
+    private String title;
+    private String content;
+    private List<CommentResponseDto> comments;
+    private MemberResponseDto member;
```
```suggestion
    private List<CommentResponseDto> commentResponseDtoList;
    private MemberResponseDto memberResponseDto;
```

inzulmi132 replied:
필드 이름이 명확하면 좋을 거 같습니다.

inzulmi132 commented on src/main/java/com/sparta/newsfeedproject/domain/post/repository/PostRepository.java:
```diff
@@ -4,4 +4,8 @@
 
 public interface PostRepository extends JpaRepository<Post, Long> {
+
+    default Post findByPostId(Long id){
+        return findById(id).orElseThrow(() -> new IllegalArgumentException("Post not found with id: " + id));
```
Repository는 그냥 리턴만 하고 예외 처리는 Service에서 하는 게 좋다고 생각합니다.

inzulmi132 commented on src/main/java/com/sparta/newsfeedproject/domain/post/service/PostService.java:
```diff
@@ -0,0 +1,24 @@
+
+    private final PostRepository postRepository;
+
+    @Transactional
+    public PostResponseDto createPost(PostRequestDto requestDto, Member member) {
+        Post post = new Post(requestDto, member);
+        postRepository.save(post);
+        return new PostResponseDto(post);
+    }
```
@Transactional이 필요한지 잘 모르겠습니다.

coldrice99 committed:
refactor: 필드 이름 변경 및 불필요한 코드 삭제

- comments를 commentResponseDtoList로 변경
- member를 memberResponseDto로 변경
- 사용하지 않는 코드 몇 개 삭제

Refs: #22

JANG-MINWOO reviewed (approved):
LGTM
